sources/engine/Stride.UI/Panels/UniformGrid.cs

```diff
@@ -84,19 +84,49 @@ public int Layers
             }
         }
 
+        /// <summary>
+        /// Calculates the total size for spanned cells including gaps.
+        /// </summary>
+        /// <param name="cellSize">The size of one cell</param>
+        /// <param name="span">The number of cells spanned</param>
+        /// <param name="dimension">The dimension (0=Column, 1=Row, 2=Layer)</param>
+        /// <returns>The total size including gaps</returns>
+        private float CalculateSpannedSize(float cellSize, int span, int dimension)
+        {
+            if (span <= 1) return cellSize;
+            
+            // For spanned elements: (cellSize * span) + (gap * (span - 1))
+            var gapSize = GetGapForDimension(dimension) * (span - 1);
+            return cellSize * span + gapSize;
+        }
+
         protected override Vector3 MeasureOverride(Vector3 availableSizeWithoutMargins)
         {
-            // compute the size available for one cell
+            // compute the size available for one cell, accounting for gaps
             var gridSize = new Vector3(Columns, Rows, Layers);
-            var availableForOneCell = new Vector3(availableSizeWithoutMargins.X / gridSize.X, availableSizeWithoutMargins.Y / gridSize.Y, availableSizeWithoutMargins.Z / gridSize.Z);
+            var totalGapSize = new Vector3(
+                CalculateTotalGapSize(0, Columns),
+                CalculateTotalGapSize(1, Rows),
+                CalculateTotalGapSize(2, Layers));
+            
+            var availableForCells = availableSizeWithoutMargins - totalGapSize;
+            var availableForOneCell = new Vector3(
+                availableForCells.X / gridSize.X, 
+                availableForCells.Y / gridSize.Y, 
+                availableForCells.Z / gridSize.Z);
```

Comment:
`CalculateTotalGapSize` could take and return vectors, and the per component division could be a vector / vector